Division by Zero error parsing a jacoco report.

Maven 3 
plugin 1.1.0


### Operating System (Ubuntu 15.10 / MacOS 10.10 / Windows 10)
Windows 10
Use `uname -a` if on Linux.

### JDK (Oracle 1.8.0_72, OpenJDK 1.8.x, Azul Zing)
1.8.0.161
Paste the output from `java -version` at the command line.

### Library Dependencies

If this is an issue that involves integration with another system, include the exact version and OS of the other system, including any intermediate drivers or APIs i.e. if you connect to a PostgreSQL database, include both the version / OS of PostgreSQL and the JDBC driver version used to connect to the database.

### Expected Behavior
Upload report
Please describe the expected behavior of the issue, starting from the first action.

1.Run coverage phase
2.
3.

### Actual Behavior
Build Failure due to / by zero error

Please provide a description of what actually happens, working from the same starting point.

Be descriptive: "it doesn't work" does not describe what the behavior actually is -- instead, say "when sending the coverage with the command (...) it returns the output error (...)"

1.
2.
3.

### Reproducible Test Case

Please provide a some information on how to reproduce the bug. A PR with a failing test would be awesome, if possible.  

StackTrace:
`[INFO] Parsing coverage data... \target\site\jacoco\jacoco.xml
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 15.979 s
[INFO] Finished at: 2019-02-08T13:01:28+10:00
[INFO] Final Memory: 14M/179M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.gavinmogan:codacy-maven-plugin:1.1.0:coverage (default-cli) on project Minigames: Execution default-cli of goal com.gavinmogan:codacy-maven-plugin:1.1.0:coverage failed: / by zero -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal com.gavinmogan:codacy-maven-plugin:1.1.0:coverage (default-cli) on project Minigames: Execution default-cli of goal com.gavinmogan:codacy-maven-plugin:1.1.0:coverage failed: / by zero
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:212)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:863)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:199)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
	at org.codehaus.classworlds.Launcher.main(Launcher.java:47)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution default-cli of goal com.gavinmogan:codacy-maven-plugin:1.1.0:coverage failed: / by zero
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:145)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:207)
	... 21 more
Caused by: java.lang.ArithmeticException: / by zero
	at com.codacy.parsers.implementation.JacocoParser.com$codacy$parsers$implementation$JacocoParser$$lineCoverage(JacocoParser.scala:47)
	at com.codacy.parsers.implementation.JacocoParser$$anonfun$5$$anonfun$apply$1.apply(JacocoParser.scala:32)
	at com.codacy.parsers.implementation.JacocoParser$$anonfun$5$$anonfun$apply$1.apply(JacocoParser.scala:30)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:245)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:245)
	at scala.collection.Iterator$class.foreach(Iterator.scala:750)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1202)
	at scala.collection.IterableLike$class.foreach(IterableLike.scala:72)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
	at scala.collection.TraversableLike$class.map(TraversableLike.scala:245)
	at scala.collection.AbstractTraversable.map(Traversable.scala:104)
	at com.codacy.parsers.implementation.JacocoParser$$anonfun$5.apply(JacocoParser.scala:29)
	at com.codacy.parsers.implementation.JacocoParser$$anonfun$5.apply(JacocoParser.scala:27)
	at scala.collection.TraversableLike$$anonfun$flatMap$1.apply(TraversableLike.scala:252)
	at scala.collection.TraversableLike$$anonfun$flatMap$1.apply(TraversableLike.scala:252)
	at scala.collection.Iterator$class.foreach(Iterator.scala:750)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1202)
	at scala.collection.IterableLike$class.foreach(IterableLike.scala:72)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
	at scala.collection.TraversableLike$class.flatMap(TraversableLike.scala:252)
	at scala.collection.AbstractTraversable.flatMap(Traversable.scala:104)
	at com.codacy.parsers.implementation.JacocoParser.generateReport(JacocoParser.scala:26)
	at com.gavinmogan.CodacyCoverageReporterMojo.processReport(CodacyCoverageReporterMojo.java:221)
	at com.gavinmogan.CodacyCoverageReporterMojo.execute(CodacyCoverageReporterMojo.java:136)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
	... 22 more
[ERROR] 
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException

Process finished with exit code 1
`

--- a/src/main/scala/com/codacy/parsers/implementation/CoberturaParser.scala
+++ b/src/main/scala/com/codacy/parsers/implementation/CoberturaParser.scala
@@ -48,7 +48,7 @@ object CoberturaParser extends CoverageParser {
       val totalValue = TextUtils.asFloat(total.text)
       (totalValue * 100).toInt
     }
-    val fileHit = classHit.sum / classHit.length
+    val fileHit = if (classHit.length != 0) { classHit.sum / classHit.length } else 0
 
     val lineHitMap: Map[Int, Int] =
       (for {

--- a/src/main/scala/com/codacy/parsers/implementation/JacocoParser.scala
+++ b/src/main/scala/com/codacy/parsers/implementation/JacocoParser.scala
@@ -68,10 +68,10 @@ object JacocoParser extends CoverageParser {
       case counter if (counter \ "@type").text == "LINE" =>
         val covered = TextUtils.asFloat((counter \ "@covered").text)
         val missed = TextUtils.asFloat((counter \ "@missed").text)
-        ((covered / (covered + missed)) * 100).toInt
+        (if ((covered + missed) > 0) (covered / (covered + missed)) * 100 else 0f).toInt
     }
 
-    val fileHit = if (lineHit.sum > 0) { lineHit.sum / lineHit.length } else 0
+    val fileHit = if (lineHit.sum != 0) { lineHit.sum / lineHit.length } else 0
 
     val lineHitMap: Map[Int, Int] = (fileNode \\ "line")
       .map { line =>

